### DISCUSSION TITLE
How to improve screen update time?

### DISCUSSION THREAD

**Discussion by jankovalski** [Bodmer/TFT_eSPI#2063] [Q&A - General]
I'm running 4.0" ST7796 with Teensy 4.1, 80 MHz SPI. It's working great and now I'm looking into optimisation.

A single execution of `tft.drawNumber(88888,240,28,6);` takes 3 ms. I draw 10 numbers like that so the total time of my LCD function is 30 ms. Is this time normal and can I lower it?

I know I could add a condition only to update the numbers that actually changed but I expect all of them to change simultaneously so it's not helpful for me.

**Comment by Bodmer** [comment]
The library is not optimised for the Teensy, so it uses the slower Arduino board package calls, hence the 10x slower speed.  For comparison an ESP32 at 80MHz takes 0.3ms to draw Font 6 digit so some significant optimisation would be possible.

I was given a Teensy 3 board 2 years ago but I have not even powered it up. I tend to use the RP2040 for projects due to the excellent PIO capability.

There are Teensy optimised TFT libraries around but I have not tried them, it might be possible to port over the optimisations but this is not something I have time to do or to support.

**Comment by jankovalski** [comment]
ESP32 is indeed the perfect choice for TFT. I wanted to try Teensy because of native SD and CAN bus which is no so perfect on ESP32... The choice of MCU for a car-dash-datalogger device is not easy.
